core/src/main/java/org/kiwix/kiwixmobile/core/search/SearchFragment.kt

```diff
@@ -100,6 +93,12 @@ class SearchFragment : BaseFragment() {
 
   override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
     super.onViewCreated(view, savedInstanceState)
+    searchAdapter = SearchAdapter(
+      RecentSearchDelegate(::onItemClick, ::onItemClickNewTab) {
+        searchViewModel.actions.offer(OnItemLongClick(it))
+      },
+      ZimSearchResultDelegate(::onItemClick, ::onItemClickNewTab)
+    )
```

Comment:
We should create `searchAdapter` lazily.

Reply:
Ok @MohitMaliDeveloper . There is one solution to hack [this](https://stackoverflow.com/questions/50173899/null-property-when-accessing-lazy-non-nullable-property-of-android-sensor). But I think it's too much engineering . What do you think?